node/impl/full/eth.go

```diff
@@ -1270,6 +1325,53 @@ func (e *EthEventHandler) EthGetLogs(ctx context.Context, filterSpec *ethtypes.E
 	return ethFilterResultFromEvents(ctx, ces, e.SubManager.StateAPI)
 }
 
+func (e *EthEventHandler) waitForHeightProcessed(ctx context.Context, height abi.ChainEpoch) error {
+	ei := e.EventFilterManager.EventIndex
+	if height > e.Chain.GetHeaviestTipSet().Height() {
+		return xerrors.New("height is in the future")
+	}
+
+	ctx, cancel := context.WithTimeout(ctx, eventReadTimeout)
+	defer cancel()
+
+	// if the height we're interested in has already been indexed -> there's nothing to do here
+	b, err := ei.IsHeightProcessed(ctx, uint64(height))
+	if err != nil {
+		return xerrors.Errorf("failed to check if event index has events for given height: %w", err)
+	}
+	if b {
+		return nil
+	}
```

Comment:
could compact this and the duplicate block below a little

```suggestion
	if b, err := ei.IsHeightProcessed(ctx, uint64(height)); err != nil {
		return xerrors.Errorf("failed to check if event index has events for given height: %w", err)
	} else if b {
		return nil
	}
```

(up to you tho)

Reply:
Reduced it below at multiple places . Thanks.